src/seismometer/core/nbhost.py

```diff
@@ -0,0 +1,22 @@
+from enum import Enum
+
+import psutil
+
+
+class NotebookHost(Enum):
+    JUPYTER_LAB = 1
+    JUPYTER_HUB = 2
+    VSCODE = 3
+    OTHER = -1
+
+    @classmethod
+    def get_current_host(cls):
+        cmds = [x for x in psutil.Process().parent().cmdline()]
```

Comment:
How close could you get from a environment variable approach?  
Looking across at variables with "J" or "VS" (VSCODE_*) seems promising:   
['JPY_SESSION_NAME', 'JPY_PARENT_PID']  

Also, taking inspiration from tqdm autodetect (CLI vs GUI), they do a try around 
`get_ipython = sys.modules['IPython'].get_ipython` for checking ipython setup without importing.


jupyterhub has a lot of options JUPYTERHUB_ variables

Reply:
FINC